build/gen_stub.php

```diff
@@ -59,6 +59,23 @@ function processStubFile(string $stubFile, Context $context): ?FileInfo {
             echo "Saved $arginfoFile\n";
         }
 
+        if ($context->generateMethodsynopses && $fileInfo->generateMethodsynopses) {
```

Comment:
Why do we want the stub to specify `@generate-methodsynopses` as well?

Reply:
I added this to be able to granularly control which methodsynopses to generate... Although, I forgot that we can already define this via the path argument... Although, using the file-level phpdoc is maybe still more convenient until the development of this functionality is finished (i.e. if the format of the generated XML changes, I don't have to run gen_stub for all the chosen directories.

Reply:
I would suggest to drop this, so we can already merge the implemented part (without the generated files of course).